pkg/controller/sparkapplication/controller.go

```diff
@@ -455,20 +455,19 @@ func shouldRetry(app *v1beta2.SparkApplication) bool {
 		}
 	case v1beta2.PendingSubmissionState:
 		var interval int64 = 257
-		if app.Spec.Mode == v1beta2.ClientMode && hasRetryIntervalPassed(&interval, app.Status.SubmissionAttempts, app.CreationTimestamp) && app.Status.SubmissionAttempts < 14 {
+		if app.Spec.Mode != v1beta2.ClusterMode && hasRetryIntervalPassed(&interval, app.Status.SubmissionAttempts, app.CreationTimestamp) && app.Status.SubmissionAttempts <= 14 {
```

Comment:
let's not hardcode 14 and use the field set as part of CRD Spec.  Defaulting if not set  in CRD is fine.

Reply:
okay i can set to submissionRetries but if its not in crd it will break the operator

Reply:
You can default in https://github.com/lyft/spark-on-k8s-operator/blob/master/pkg/apis/sparkoperator.k8s.io/v1beta2/defaults.go to be safe ?

Reply:
+1

Reply:
okay sounds good, added

Reply:
going to test in flyte staging before merging this

Reply:
tested : https://flyte-staging.lyft.net/console/projects/flytefunctionaltests/domains/development/executions/qfuyx7ac8m